drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -719,6 +844,12 @@ static int esp32_wifi_stats(const struct device *dev, struct net_stats_wifi *sta
 static int esp32_wifi_dev_init(const struct device *dev)
 {
 	esp_timer_init();
+	wifi_init_config_t config = WIFI_INIT_CONFIG_DEFAULT();
+	esp_err_t ret = esp_wifi_init(&config);
+
+	if (ret != ESP_OK) {
+		LOG_ERR("Enable to initialize the wifi");
```

Comment:
In case wifi init fail, maybe call `esp_wifi_deinit()` to free resources.
Also typo: "Enable -> Unable"

Reply:
>In case wifi init fail, maybe call esp_wifi_deinit() to free resources.

if `esp_wifi_init(&config);` fails to initialise then it calls `esp_wifi_deinit()` internally to free resources. For reference,

```
esp_err_t esp_wifi_init(const wifi_init_config_t *config)
{
	esp_wifi_power_domain_on();

#if CONFIG_ESP32_WIFI_SW_COEXIST_ENABLE
	coex_init();
#endif

#if CONFIG_IDF_TARGET_ESP32
	s_wifi_mac_time_update_cb = esp_wifi_internal_update_mac_time;
#endif

    esp_wifi_internal_set_log_level(CONFIG_WIFI_LOG_LEVEL);

	esp_err_t result = esp_wifi_init_internal(config);
	if (result == ESP_OK) {
		esp_phy_modem_init();
		result = esp_supplicant_init();
		if (result != ESP_OK) {
		LOG_ERR("Failed to init supplicant (0x%x)", result);
			esp_wifi_deinit();
		}
	}
    adc2_cal_include(); //This enables the ADC2 calibration constructor at start up.

	return result;
}
``` 
So, I think there is no need to deinit it again. 

>In case wifi init fail, maybe call esp_wifi_deinit() to free resources.
Also typo: "Enable -> Unable"

Let me make this correction.